src/core/public/saved_objects/saved_objects_client.ts

```diff
@@ -227,6 +234,11 @@ export class SavedObjectsClient {
     this.batchQueue = [];
   }
 
+  public setCurrentWorkspace(workspaceId: string): boolean {
```

Comment:
why do we need to have a separate set workspace id when using saved object client? Can workspaces field be within the options? If each workspace should have one saved object client instance, we can add workspace id in the constructor

Reply:
This method is used in workspace.start lifeCycle, which indicates current workspace id. We support workspaces field within options, but it is import that we add a internal `currentWorkspaceId` so that no matter where calls savedObjectsClient methods, the client will append workspaces fields when doing the real request.

Reply:
What do you mean to make the workspaces transparent? this is to set the workspace id not get, right? plus, why would plugins need to get current workspace id from the saved objects client. Can we have the workspace id in the global context instead of setting it specific for saved object client, is it the only place where it needs to know the current workspace?

Reply:
The `setCurrentWorkspace` will only be called by workspace plugin when setup, and once the `currentWorkspaceId` is set, the workspaces field will be automatically append into the final request payload accordingly.